macro/headers.py

```diff
@@ -406,35 +405,27 @@ def get_badges(macro, data):
     html = _render_badges(macro, data, badges)
     return html
 
-
-def _process_haros_data(data):
-    haros = []
-    if data.get('haros_data', []):
-        haros_data = data.get('haros_data', [])
- 
-        type = ""
-        url = ""
-        for datum in haros_data:
-            if "report" in datum:
-                type = datum['report'].get('type')
-                url = datum['report'].get('url')
-                haros.append({'type' : str(type), 'url' : str(url)})
-    return haros
-
-def _render_haros(macro, data, haros):
-    p = macro.formatter.paragraph
+def _render_reports(macro, reports):
+    # p = macro.formatter.paragraph
     html = ''
-
-    if haros:
+    if reports and len(reports)>0:
```

Comment:
Sorry I don't understand.
If `reports` is `None`, then `len(reports)` would crash, so we must first check for that.
When `reports` is a list of length 0, we should not print anything either.

Reply:
In Python that is actually not the case @max-krichenbauer.

Both `reports==None` and `len(reports)==0` will resolve to `False`. There is no need to check for those separately.

Reply:
I've just tested it here and

```python
reports=None
if len(reports)>0:
   [...]
```

positively crashes with 

```
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    if len(reports)>0:
TypeError: object of type 'NoneType' has no len()
```

Is this a new feature of newer Python versions and if so, which version of Python is the Wiki guaranteed to run on?

Reply:
This has nothing to do with Python versions.

```python
if reports:
    ...
```

will not crash. It will check whether `reports` is `None` *and* whether `len(reports)` is `> 0`.

Reply:
Ah, now I understand. I though the suggestion was to remove `if reports`.
To be honest, I'm surprised and appalled by this behavior of Python, but it's not up to me to judge Python design here, so I changed it as requested.